force-app/main/default/classes/SOQL.cls

```diff
@@ -286,24 +405,49 @@ public with sharing class SOQL {
      * @param A set of Lead Ids.
      * @return A list of Accounts with matching email domains.
      */
-    public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
-        // Query the Leads
+    public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) 
+    {
+        // Query the Leads to get their emails
+        List<Lead> leads = [SELECT Email FROM Lead WHERE Id IN :leadIds];
 
-        // Create a set of email domains from the Leads
+        // Create a set to hold email domains from the Leads
+        Set<String> emailDomains = new Set<String>();
 
         // Loop through each Lead
-
-            // Get the email from the Lead
-
-            // Check if the email is not null
-
-                // Add the email domain to the set
-
-        
-        // Query the Accounts
-        return null; // Replace null with the variable you used to store the result
+    for (Lead lead : leads) 
+    {
+        // Get the email from the Lead
+        String email = lead.Email;
+
+        // Check if the email is not null
+        if (email != null) 
+        {
+        // Extract the domain from the email
+        String domain = email.substring(email.indexOf('@') + 1);
+        emailDomains.add(domain);
+        }
+    }
+        // Query the Accounts with matching website domains
+        List<Account> accounts = [SELECT Id, Name, Website FROM Account WHERE Website != null];
```

Comment:
It is probably not good to query all of the accounts and then loop through them. You are likely to hit limits with large data volumes this way. Instead, you have a list of email domains above, why not use that as an input for the query?